storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2905,6 +2929,10 @@ public void launchServer() throws Exception {
                                             throw new RuntimeException(e);
                                         }
                                     });
+
+            //Should we make the delaySecs and recurSecs in sync with any conf value?
```

Comment:
Why not update the metricset active flag when Nimbus becomes active instead of polling for changes?

Reply:
I would very much love to. However based on current Nimbus implementation there's no way to do it. There's is however one leadership change callback buried in Zookeeper side code (See `LeaderListenerCallback`), which communicates through IStormClusterState. It'll be great if you can open a PR to improve Nimbus in general, along with [STORM-3187](https://issues.apache.org/jira/browse/STORM-3187). Otherwise we kind of have to using busy waiting here.

Reply:
Okay, we can do it later. Filed https://issues.apache.org/jira/browse/STORM-3192. Please remove the comment since it seems like it wouldn't be useful outside this PR.